### PR TITLE
clickhouse array support

### PR DESCRIPTION
**Pull request by gane5hvarma**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-transformer/534)

### CODE REVIEW

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: v0/destinations/clickhouse/transform.js [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+const { validTimestamp } = require("../../../warehouse/util");
 
 const clickhouse = "clickhouse";
 
 function processSingleMessage(message, options) {
   return processWarehouseMessage(message, options);
 }
 
-function getDataTypeOverride(val, options) {}
+function getDataType(val) {
```
Comment (psrikanth88) [inline v0/destinations/clickhouse/transform.js:10]: @gane5hvarma can't we use the function from warehouse/index.js

File: v0/destinations/clickhouse/transform.js [hunk 2]
```diff
@@ -1,12 +1,58 @@
...
+        if (finalDataType === "string") {
+          break;
+        }
+        if (dataType === "float" && finalDataType === "int") {
+          finalDataType = "float";
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        if (dataType === "int" && finalDataType === "float") {
+          finalDataType = "float";
```
Comment (psrikanth88) [inline v0/destinations/clickhouse/transform.js:45]: remove this line as its redundant?

File: v0/destinations/clickhouse/transform.js [hunk 3]
```diff
@@ -1,12 +1,58 @@
...
+      return "string";
+    }
+    // check for different data types in the array. if there are different then return array(string)
+    const firstValueDataType = getDataType(val[0]);
+    let finalDataType = firstValueDataType;
+    for (let i = 1; i < val.length; i += 1) {
+      const dataType = getDataType(val[i]);
+      if (finalDataType !== dataType) {
+        if (finalDataType === "string") {
+          break;
```
Comment (psrikanth88) [inline v0/destinations/clickhouse/transform.js:37]: don't we need to do .toString for other types. Will clickhouse handle that

**Review by psrikanth88 — CHANGES_REQUESTED** [review]

File: v0/destinations/clickhouse/transform.js (hunk 3 above)
Comment (gane5hvarma) [reply v0/destinations/clickhouse/transform.js:37]: yes, clickhouse.go is handling this

**Review by sivashanmukh — APPROVED** [review]
LGTM

**Review by psrikanth88 — APPROVED** [review]

**Review by psrikanth88 — APPROVED** [review]

**Pull request merged**